File: asteroids/Components/BulletMovement.cs
using SlimMath;
using Vortex.Scenegraph.Components;

namespace asteroids.Components
{
    public class BulletMovement : ScriptComponent
    {
        public Vector3 MovementVector { get; set; }
        public float MovementSpeed { get; set; }

        public override void OnUpdate(float delta)
        {
            base.OnUpdate(delta);
            Entity.LocalPosition += (MovementVector * MovementSpeed * delta);
        }
    }
}

File: Data/bulletMovement.js
/// <reference path="scriptDefines.js" />

var MovementVector = new SlimMath.Vector3(0, 0, 0);

var MovementSpeed = 0;

function onUpdate(delta) {

    var mv = SlimMath.Vector3.Add(
        entity.TransformComponent.LocalPosition,
        SlimMath.Vector3.Multiply(MovementVector, MovementSpeed * delta));

    entity.TransformComponent.LocalPosition = mv;
}
